java/bulk-export/bulk-export-core/src/test/java/sleeper/bulkexport/model/BulkExportLeafPartitionQuerySerDeTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.bulkexport.model;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class BulkExportLeafPartitionQuerySerDeTest {
+
+    private final BulkExportLeafPartitionQuerySerDe querySerDe = new BulkExportLeafPartitionQuerySerDe();
+
+    @Test
+    public void shouldSerDeBulkExportLeafPartitionQuery() {
+        // Given
+        BulkExportLeafPartitionQuery bulkExportLeafPartitionQuery = BulkExportLeafPartitionQuery.builder()
+                .tableId("t1")
+                .exportId("e1")
+                .subExportId("se1")
+                .leafPartitionId("lp1")
+                .files(Collections.singletonList("/test/file.parquet"))
+                .build();
+
+        // When
+        String serialisedQuery = querySerDe.toJson(bulkExportLeafPartitionQuery);
+
+        BulkExportLeafPartitionQuery deserialisedQuery = querySerDe.fromJson(serialisedQuery);
+
+        // Then
+        assertThat(bulkExportLeafPartitionQuery).isEqualTo(deserialisedQuery);
```

Comment:
Please can we assert on the actual JSON produced? It would be good for something to fail if we accidentally make a backwards-incompatible change.

One option would be to use Approvals.verify, as in CompactionJobIdAssignmentCommitRequestSerDeTest.